src/client.js

```diff
@@ -0,0 +1,124 @@
+import debug from 'debug'
+import graphql from 'graphql'
+
+const log = debug('gql-ws').extend('client')
+const {
+  stripIgnoredCharacters, parse,
+} = graphql
+const NORMAL_CLOSURE = 1000
+
+export default (PassThrough, WebSocket) =>
+  class Client {
+    // variables are initialized because they are private
+    // awaiting https://github.com/evanw/esbuild/issues/47
+    // to use the # notation
+    ws
+    operation_id = 0
+    address
+
+    /**
+     * Create a graphql client
+     * @param {String} address The ws connection uri.
+     */
+    constructor(address) {
+      if (typeof address !== 'string')
+        throw new TypeError(`The address must be a string`)
+      this.address = address
+    }
+
+    /**
+     * Connect the client to his address,
+     * a client must be disconnected before connecting again
+     * or it will throw an error
+     */
+    async connect() {
+      this.ws = new WebSocket(this.address)
+      await new Promise((resolve, reject) => {
+        this.ws.addEventListener('open', resolve)
+        this.ws.addEventListener('error', reject)
+      })
+      log('client ready')
+    }
+
+    disconnect() {
+      if (!this.ws)
+        throw new Error('You must connect before disconnecting u genius..')
+      this.ws.close(NORMAL_CLOSURE, 'closed by client')
+      log('disconnected')
+    }
+
+    async query(query, variables = {}) {
+      if (typeof query !== 'string')
+        throw new Error('The query is not a String')
+
+      // fail fast
+      parse(query)
```

Comment:
You really need to import the whole graphql parser just for this ?

Reply:
removed